lib/puma/server.rb

```diff
@@ -492,22 +492,21 @@ def with_force_shutdown(client, &block)
     # :nocov:
 
     # Handle various error types thrown by Client I/O operations.
-    def client_error(e, client)
+    def client_error(e, client, requests = 1)
       # Swallow, do not log
       return if [ConnectionError, EOFError].include?(e.class)
 
-      lowlevel_error(e, client.env)
       case e
       when MiniSSL::SSLError
         @log_writer.ssl_error e, client.io
```

Comment:
Looks like your `lowlevel_error_handler` (if you had one) was called for `MiniSSL::SSLError` before but now it is not. Is that a change we want to make?

Reply:
we don't seem to want to sent anything back to client for this error right? because there is no `client.write_error` for it 🤔

Reply:
We don't want to send anything, but we still want to call the low-level error handler.

Reply:
I think we didn't send anything back in this case because an ssl error usually means we didn't even establish a connection, so there's probably no client to send anything back to. But we absolutely still want to call the error handler, as someone may have a side effect on their error handler to do something like call Sentry or some other error notification service.

Reply:
So I guess: we could try to send something back to the client but just not care if it fails?

Reply:
Thanks for the clarification @nateberkopec. I think it makes sense that we need to call the handler, so I moved the call to the `MiniSSL::SSLError` case